server/src/main/java/org/elasticsearch/bootstrap/Elasticsearch.java

```diff
@@ -326,10 +326,6 @@ public boolean handle(int code) {
             // we've already logged this.
         }
 
-        Natives.trySetMaxNumberOfThreads();
```

Comment:
was this previously was 'caching' maxNumberOfThreads, MaxSizeVirtualMemory etc?
Do we want to do something similar with this refactoring?

Reply:
Previously these try methods were called here, which set the values returned from rlimit in global statics in JNANatives. Now these values are retrieved during construction of NativeAccess instead of explicitly. By doing it during construction they do not need to be mutable as they were before.